llmfoundry/utils/config_utils.py

```diff
@@ -178,9 +183,102 @@ def log_config(cfg: DictConfig) -> None:
             wandb.config.update(om.to_container(cfg, resolve=True))
 
     if 'mlflow' in cfg.get('loggers', {}):
-        try:
-            import mlflow
-        except ImportError as e:
-            raise e
+        if not mlflow:
+            raise ImportError('MLflow is required but not installed.')
         if mlflow.active_run():
             mlflow.log_params(params=om.to_container(cfg, resolve=True))
+            _log_dataset_uri(cfg)
+
+
+def _parse_source_dataset(cfg: DictConfig) -> Set[Tuple[str, str, str]]:
+    """Parse a run config for dataset information.
+
+    Given a config dictionary, parse through it to determine what the datasource
+    should be catagorized as. Possible data sources are Delta Tables, UC Volumes,
+    HuggingFace paths, remote storage, or local storage.
+
+    Args:
+        cfg (DictConfig): A config dictionary of a run
+
+    Returns:
+        Set[Tuple[str]]: A set of tuples formatted as (data type, path, split)
+    """
+    data_paths = set()
+
+    for data_split in ['train', 'eval']:
+        data_set = cfg.get(f'{data_split}_loader', {}).get('dataset', {})
+        cfg_split = data_set.get('split', None)
+        source_dataset_path = cfg.get(f'source_dataset_{data_split}', {})
+
+        # Check for Delta table
+        if source_dataset_path and len(source_dataset_path.split('.')) >= 3:
+            data_paths.add(('delta_table', source_dataset_path, data_split))
+        # Check for UC volume
+        elif source_dataset_path and source_dataset_path.startswith('dbfs:'):
+            data_paths.add(('uc_volume', source_dataset_path[len('dbfs:'):], data_split))
+        # Check for HF path
+        elif 'hf_name' in data_set:
+            hf_path = data_set['hf_name']
+            backend, _, _ = parse_uri(hf_path)
+            if backend:
+                hf_path = f'{hf_path.rstrip("/")}/{cfg_split}' if cfg_split else hf_path
+                data_paths.add((backend, hf_path, data_split))
+            elif hf_path.startswith('/'):
+                data_paths.add(('local', hf_path, data_split))
+            else:
+                data_paths.add(('hf', hf_path, data_split))
+        # check for remote path
+        elif 'remote' in data_set:
+            remote_path = data_set['remote']
+            backend, _, _ = parse_uri(remote_path)
+            if backend:
+                remote_path = f'{remote_path.rstrip("/")}/{cfg_split}/' if cfg_split else remote_path
+                data_paths.add((backend, remote_path, data_split))
+            else:
+                data_paths.add(('local', remote_path, data_split))
+        else:
+            raise KeyError('DataSource Not Found')
```

Comment:
lets not error here. a bug in this code doesn't need to take down the entire job. lets just log a warning